.github/workflows/cy.yml

```diff
@@ -0,0 +1,29 @@
+name: CI
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  push_to_ghcr:
+    runs-on: ubuntu-latest
+    if: github.event_name == 'push' && github.ref == 'refs/heads/main'
```

Comment:
If this is the only job then we could remove `pull_request:` and the `if` here and it'll only run on pushes to main.

I'd suggest giving it a test as well without this so we can see it working.